### ISSUE TITLE
Incorrect "did you mean to call" error issued for invoked class property

### ISSUE THREAD

**Issue by RyanCavanaugh** [microsoft/TypeScript#38161]
**TypeScript Version:**  nightly

**Search Terms:** "did you mean to call"

**Code**

```ts
declare const variable: () => void;
class Foo {
  method!: (() => void);

  f() {
    // Error (incorrect)
    if (this.method) {
      this.method();
    }

    // OK (correct)
    if (variable) {
      variable();
    }
  }

}
```

**Expected behavior:** No error

**Actual behavior:** Error: This condition will always return true; did you mean to call it instead?

**Playground Link:** [Linky](https://www.typescriptlang.org/play/?ts=3.9.0-dev.20200423&ssl=1&ssc=1&pln=18&pc=1#code/CYUwxgNghgTiAEYD2A7AzgF3gN1gSygCMIQAueACgEp4BeAPhyT2AG4AoSKNNeAMSRJ4Ab3bx4AWxAYAFkmABCchWp1G2ZsCocx8AGarR48QHoT8AKIwYSGLvF49lWXjQA6KbPk0jx8S-dPOWBqDj8AX3Z7eDN4AHkAaWjHSlwYAmIQH2jxNIySUOjI8Uj2SKA)

**Related Issues:**  Feature introduced at PR #37152

**Comment by decherneyge** [comment]
im not 100% sure this is the same issue but I thought I would check here first before opening a separate issue. 

This image is from the typescript playground. Im adding it to show that the compiler is barking at the  null check if/when the property is not used within the block but not when it is used which seems inconsistent. The error in both places shown with a red underline is "This condition will always return true since the function is always defined. Did you mean to call it instead?(2774)"
![image](https://user-images.githubusercontent.com/654448/84954949-54bd6b80-b0c4-11ea-95cf-e11d441d41c0.png)

![image](https://user-images.githubusercontent.com/654448/84955325-196f6c80-b0c5-11ea-9ff8-78a2d7c1ce63.png)


Code:
```
class Widget {
    public refresh: (() => void);
    constructor() {
        this.refresh = () => { };
    }
}

class Main{

    public widgets: Array<Widget> = [];
    public $timeout: any;

    constructor() {

        const filterable: Array<Widget> = this.widgets.filter((x) => {
            //error here
            if (x.refresh) {
                //x.refresh();
            }
            //no error here
            if (x.refresh) {
                x.refresh();
            }
        });
        
    }

    private refreshWidget(w: Widget) {
        //error here
        if (w.refresh) {
            this.$timeout(() => {
                //w.refresh();
             });
        }
        //no error here
        if (w.refresh) {
            this.$timeout(() => {
                w.refresh();
             });
        }

    }

}

**Comment by RyanCavanaugh** [comment]
@decherneyge that is precisely the intended behavior

**Comment by decherneyge** [comment]
@RyanCavanaugh oh ok. Thanks for the reply! Why is it the intended behavior?

**Comment by RyanCavanaugh** [comment]
@decherneyge See discussion in https://github.com/microsoft/TypeScript/pull/32802 and linked issues from there

**Comment by luryson** [comment]
```javascript
interface A {
        a: () => boolean;
    }

    class AImpl implements A {
        public a(): boolean {
            return true;
        }
    }

    it('should fail', function () {
        const aImpl = new AImpl();
        if (aImpl.a) { // TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?
            console.log("always true");
        }
    });

    it('why success', function () {
        const aImpl = new AImpl();
        if (!aImpl.a) { // nothing happens here
            console.log('always false');
        }
    });
```

just wondering why `!aIpl.a` not bar baring TS2774
